.github/workflows/hosted-runner.yaml

```diff
@@ -0,0 +1,93 @@
+name: Test Self hosted runner addition
+
+on:
+  workflow_call:
```

Comment:
We will need a strategy based on workflow_run to get access to secrets and test the binary build from the forked repo, with the current build you have here it will build it form the head of the main instead of from the PR

Reply:
Similar to https://github.com/crc-org/ci-definitions/blob/70d016f5fd15e7cbdf8e3056c500654b0196a049/.github/workflows/crc-builder-pusher.yml#L18 stragtey for build / push

Reply:
we also have to store the backend in a cloud storage to be able to destroy the created resources later

Reply:
@anjannath  ^^ Just added all required secrets for this:

- ARM_TENANT_ID
- ARM_SUBSCRIPTION_ID
- ARM_CLIENT_ID
- ARM_CLIENT_SECRET
- AZURE_STORAGE_ACCOUNT
- AZURE_STORAGE_KEY

Also to use the remote storage for backed-url the url would be something like (notice base name should exactly match `mapt-gh-runner-mapt-state`):

```
--backed-url azblob://mapt-gh-runner-mapt-state/${{ github.repository }}-${{ github.run_id }} 
```

Reply:
It seems `workflow_run` only works when the workflow is already in the `main` branch, so need to test this first on my fork and then update the PR, we'll have to initially merge something and then create follow up PR to fix if any issues found